cmd/alertmanager/main.go

```diff
@@ -185,6 +187,10 @@ func run() int {
 		retention       = kingpin.Flag("data.retention", "How long to keep data for.").Default("120h").Duration()
 		alertGCInterval = kingpin.Flag("alerts.gc-interval", "Interval between alert GC.").Default("30m").Duration()
 
+		alertStorageProvider = kingpin.Flag("alerts.storage.provider", "Alert storage \"mem\" or \"etcd\".").Default("mem").String()
+		alertEtcdEndpoints   = kingpin.Flag("alerts.etcd.endpoint", "Etcd server used for alert storage(may be repeated for each in etcd cluster).").Default("localhost:2379").Strings()
```

Comment:
SHOULD support following parameters as most etcd server run on self-signed CA if they enable TLS:

```
# etcdctl -h | grep -- -file
   --cert-file value                identify HTTPS client using this SSL certificate file
   --key-file value                 identify HTTPS client using this SSL key file
   --ca-file value                  verify certificates of HTTPS-enabled servers using this CA bundle
```

Reply:
You're right.  I'll wait on this one.  Need to see how upstream maintainers react to this PR.  If they react friendly to this PR, then I'll implement those enhancements.  I don't want to invest unless I know this PR will have a chance.